[BUG] `dp test` is invalid in multitask mode

### Bug summary

```bash
dp --pt test -m model.ckpt-1000000.pt -s /home/data/wangh-v1.3/Domains/Anode/downstream/test --head Domains_Anode
[2024-03-16 09:04:59,171] DEEPMD WARNING To get the best performance, it is recommended to adjust the number of threads by setting the environment variables OMP_NUM_THREADS, DP_INTRA_OP_PARALLELISM_THREADS, and DP_INTER_OP_PARALLELISM_THREADS. See https://deepmd.rtfd.io/parallelism/ for more information.
Traceback (most recent call last):
  File "/opt/deepmd-kit-3.0.0/bin/dp", line 10, in <module>
    sys.exit(main())
  File "/opt/deepmd-kit-3.0.0/lib/python3.10/site-packages/deepmd/main.py", line 805, in main
    deepmd_main(args)
  File "/opt/deepmd-kit-3.0.0/lib/python3.10/site-packages/deepmd/entrypoints/main.py", line 64, in main
    test(**dict_args)
  File "/opt/deepmd-kit-3.0.0/lib/python3.10/site-packages/deepmd/entrypoints/test.py", line 129, in test
    dp = DeepEval(model, head=head)
  File "/opt/deepmd-kit-3.0.0/lib/python3.10/site-packages/deepmd/infer/deep_eval.py", line 295, in __new__
    deep_eval = DeepEvalBackend(
  File "/opt/deepmd-kit-3.0.0/lib/python3.10/site-packages/deepmd/pt/infer/deep_eval.py", line 104, in __init__
    assert not self.multi_task, "multitask mode currently not supported!"
AssertionError: multitask mode currently not supported!
```

### DeePMD-kit Version

3.0 alpha

### TensorFlow Version

pytorch vesion

### How did you download the software?

Offline packages

### Input Files, Running Commands, Error Log, etc.

The pretrained model is big. 

### Steps to Reproduce

```
dp --pt test -m model.ckpt-1000000.pt -s /home/data/wangh-v1.3/Domains/Anode/downstream/test --head Domains_Anode
```

### Further Information, Files, and Links

_No response_

--- a/deepmd/pt/infer/deep_eval.py
+++ b/deepmd/pt/infer/deep_eval.py
@@ -90,6 +90,7 @@ class DeepEval(DeepEvalBackend):
         *args: List[Any],
         auto_batch_size: Union[bool, int, AutoBatchSize] = True,
         neighbor_list: Optional["ase.neighborlist.NewPrimitiveNeighborList"] = None,
+        head: Optional[str] = None,
         **kwargs: Dict[str, Any],
     ):
         self.output_def = output_def
@@ -99,9 +100,24 @@ class DeepEval(DeepEvalBackend):
             if "model" in state_dict:
                 state_dict = state_dict["model"]
             self.input_param = state_dict["_extra_state"]["model_params"]
-            self.input_param["resuming"] = True
             self.multi_task = "model_dict" in self.input_param
-            assert not self.multi_task, "multitask mode currently not supported!"
+            if self.multi_task:
+                model_keys = list(self.input_param["model_dict"].keys())
+                assert (
+                    head is not None
+                ), f"Head must be set for multitask model! Available heads are: {model_keys}"
+                assert (
+                    head in model_keys
+                ), f"No head named {head} in model! Available heads are: {model_keys}"
+                self.input_param = self.input_param["model_dict"][head]
+                state_dict_head = {"_extra_state": state_dict["_extra_state"]}
+                for item in state_dict:
+                    if f"model.{head}." in item:
+                        state_dict_head[
+                            item.replace(f"model.{head}.", "model.Default.")
+                        ] = state_dict[item].clone()
+                state_dict = state_dict_head
+            self.input_param["resuming"] = True
             model = get_model(self.input_param).to(DEVICE)
             model = torch.jit.script(model)
             self.dp = ModelWrapper(model)

